File: renderer/android/src/main/java/org/scilab/forge/jlatexmath/BaseObjectHelper.java
package org.scilab.forge.jlatexmath;

import org.scilab.forge.jlatexmath.cyrillic.CyrillicRegistration;
import org.scilab.forge.jlatexmath.greek.GreekRegistration;

public class BaseObjectHelper {
	public static String getPath(Object base, String name) {
		String ret = null;
		if (base == CyrillicRegistration.class) {
			ret = "cyrillic/" + name;
		} else if (base == GreekRegistration.class) {
			ret = "greek/" + name;
		} else {
			ret = name;
		}
		return ret;
	}
}

File: renderer/ios/LatexRenderer/Classes/BaseObjectHelper.m
//
//  BaseObjectHelper.m
//  JLaTeXMath-iOS
//
//  Created by JackyKuo on 2015/6/20.
//  Copyright (c) 2015年 JackyKuo. All rights reserved.
//

#import "BaseObjectHelper.h"
#import "CyrillicRegistration.h"
#import "GreekRegistration.h"

@implementation BaseObjectHelper
+(NSString *)getPathWithNSObject:(NSObject *)base withName:(NSString *)name
{
    NSString* ret;
    if([base isKindOfClass:[RXCyrillicRegistration class]]){
        ret = [@"cyrillic/" stringByAppendingString:name];
    }else if([base isKindOfClass:[RXGreekRegistration class]]){
        ret = [@"greek/" stringByAppendingString:name];
    }else{
        ret = name;
    }
    return ret;
}
@end
